### DISCUSSION TITLE
Hooks can't be used with web components

### DISCUSSION THREAD

**Discussion by dimavs** [preactjs/preact#3536] [General]
I am getting that "@TypeError: Cannot read properties of null (reading '__H')" exception trying to use hooks inside web components. There is an example: https://plnkr.co/edit/PhINCNxhIe3QUR92

Raw preact web component is fine ('test-raw'), component with hooks rendered by name is fine too ('TestHooks'), but the same TestHooks component referred as a web component ('test-hooks') throws that exceptiion. 

What I am doing wrong?
Thanks!

PS 10.7.2 broke rendering by name (10.7.1 is fine)

**Comment by bradenmacdonald** [comment]
In case anyone else finds this after googling... this error occurs when you have multiple versions of `preact` conflicting with each other, usually because you are importing several packages from esm.sh or skypack.dev or similar, and their dependencies aren't de-duped. To fix it, use NodeJS or a bundler like rollup to install/combine your dependencies without duplicate versions of `preact`.

**Reply by rschristian** [reply]
You can use a CDN and avoid duping FWIW, but it can be a bit difficult.

Each Preact import *must* resolve to exactly the same version and you need to take care to externalize your deps ([esm.sh docs](https://esm.sh/#using-import-maps)) as much as possible.
